read-fonts/src/tables/layout/script.rs

```diff
@@ -0,0 +1,269 @@
+//! Additional support for working with OpenType scripts.
+
+use super::{ScriptList, Tag};
+use std::ops::Deref;
+
+/// A script chosen from a set of candidate tags.
+///
+/// Returned by the [`ScriptList::select`] method.
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub struct SelectedScript {
+    /// The actual OpenType tag of the chosen script.
+    pub tag: Tag,
+    /// Index of the script in the [`ScriptList`].
+    pub index: u16,
+    /// True if a script was chosen that wasn't in the requested list.
+    pub is_fallback: bool,
+}
+
+impl<'a> ScriptList<'a> {
+    /// Returns the index of the script with the given tag.
+    pub fn index_for_tag(&self, tag: Tag) -> Option<u16> {
+        self.script_records()
+            .binary_search_by_key(&tag, |rec| rec.script_tag())
+            .map(|index| index as u16)
+            .ok()
+    }
+
+    /// Finds the first available script that matches one of the given tags.
+    ///
+    /// When none of the requested scripts are available, then `DFLT`, `dflt`
+    /// and `latn` tags are tried in that order.
+    ///
+    /// If you're starting from a Unicode script code, use the
+    /// [`script_tags_from_unicode`] function to generate the appropriate set
+    /// of tags to pass to this method.
+    ///
+    /// See [`hb_ot_layout_table_select_script`](https://github.com/harfbuzz/harfbuzz/blob/2edc371e97d6d2c5ad0e085b26e9af0123501647/src/hb-ot-layout.cc#L547)
+    /// for the corresponding HarfBuzz function.
+    pub fn select(&self, tags: &[Tag]) -> Option<SelectedScript> {
+        for &tag in tags {
+            if let Some(index) = self.index_for_tag(tag) {
+                return Some(SelectedScript {
+                    tag,
+                    index,
+                    is_fallback: false,
+                });
+            }
+        }
+        for tag in [
+            // Try finding 'DFLT'
+            Tag::new(b"DFLT"),
+            // Try with 'dflt'; MS site has had typos and many fonts use it now :(
+            Tag::new(b"dflt"),
+            // try with 'latn'; some old fonts put their features there even though
+            // they're really trying to support Thai, for example :(
+            Tag::new(b"latn"),
+        ] {
+            if let Some(index) = self.index_for_tag(tag) {
+                return Some(SelectedScript {
+                    tag,
+                    index,
+                    is_fallback: true,
+                });
+            }
+        }
+        None
+    }
+}
+
+/// A prioritized list of OpenType script tags mapped from a Unicode script
+/// tag.
+///
+/// Generated by [`script_tags_from_unicode`].
+#[derive(Copy, Clone, PartialEq, Eq, Default)]
+pub struct ScriptTags {
+    tags: [Tag; 3],
+    len: usize,
+}
+
+impl Deref for ScriptTags {
```

Comment:
Optional, would an explicit call like `as_tags()` be overly horrid? - not a huge fan of Deref.

Reply:
`as_slice()` would be the idiomatic conversion fn here, but since this type is essentially an array deref to slice is totally reasonable (having both would be the most idiomatic approach)

Reply:
Agreed with Colin on both points. Added `as_slice()` for those that prefer explicit calls.